thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/ThriftProtocolFactories.java

```diff
@@ -120,7 +136,21 @@ public static TProtocolFactory binary(int maxStringLength, int maxContainerLengt
         final int maxStringLength0 = maxStringLength == 0 ? -1 : maxStringLength;
         final int maxContainerLength0 = maxContainerLength == 0 ? -1 : maxContainerLength;
 
-        // Thrift 0.9.x does not a constructor taking only maxStringLength and maxContainerLength.
+        if (IS_THRIFT_091) {
+            // Thrift 0.9.1 does not support a message length limit.
```

Comment:
I'm wondering if it is better to validate this at `ThriftSerializationFormats`.

At the very least, I think we may want to document this behavior at the following location:
https://github.com/line/armeria/blob/1179e84f06958115d0d032d605f53f8648d3ada2/thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/ThriftSerializationFormats.java#L112

Reply:
Nice suggestion. A validation close to `.protocolFactory()` would be useful. I will add an additional validation to `protocolFactory(SerializationFormat,int,int)` which will raise IllegalArgumentException if length limits are set and the Thrift version is 0.9.1.

Reply:
I just realized that a max request or response length is passed to `maxStringLength` to create the default one.
We need some refactoring to fix the default value depending on Thrift versions.
https://github.com/line/armeria/blob/a5ec2c2a0e81f20ab215f06023488b299acd2312/thrift0.13/src/main/java/com/linecorp/armeria/internal/client/thrift/THttpClientDelegate.java#L100-L106
I'm not sure we need such a change for this legacy version. Let me leave a warning log and update Javadoc's description.

Reply:
I'm fine with the warning log.